- add a clarification about recommended feature options names (#5484)
conan-io/conan-center-index: docs/reviewing.md

- use the `cmake_find_package` or `cmake_find_package_multi` generators in `test_package/conanfile.py`
- corresponding call to `find_package()` with the components _explicitly_ used in `target_link_libraries`

### Recommended feature options names

> It's often needed to add options to toggle specific library features on/off. Sometimes, it's preferred to have options enabled by default, sometimes disabled by default. In order to avoid the fragmentation, we recommend to use the following naming conventions for such options:

Reviewer: One "pillar" we've had is positive naming of options

```suggestion
It's often needed to add options to toggle specific library features on/off. Regardless of the default, there is a strong preference for using positive naming for options. In order to avoid the fragmentation, we recommend to use the following naming conventions for such options:
```

> - enable_xxx / disable_xxx

Reviewer: Typically this is in regards to "features"

I think there's a split on `enable_feature` and just `feature`

> - enable_xxx / disable_xxx

Author: ```suggestion
- enable_<feature> / disable_<feature>
```

> - with_xxx / without_xxx

Reviewer: Thus far I think the consensus is `with_dependency`

> - use_xxx

Reviewer: I've never seen a `use_xxx` and searching quickly did not produce any

```suggestion
```

Author: there are at least some usages:
```
./recipes/sqlite3/all/conanfile.py:34:        "use_alloca": [True, False],

./recipes/openblas/all/conanfile.py:21:        "use_thread": [True, False],

./recipes/ceres-solver/all/conanfile.py:17:               "use_glog":  [True, False], #TODO Set to true once gflags with nothreads=False binaries are available. Using MINILOG has a big performance drawback.
./recipes/ceres-solver/all/conanfile.py:18:               "use_gflags": [True, False],
./recipes/ceres-solver/all/conanfile.py:19:               "use_custom_blas": [True, False],
./recipes/ceres-solver/all/conanfile.py:20:               "use_eigen_sparse": [True, False],
./recipes/ceres-solver/all/conanfile.py:21:               "use_TBB": [True, False],
./recipes/ceres-solver/all/conanfile.py:22:               "use_CXX11_threads": [True, False],
./recipes/ceres-solver/all/conanfile.py:23:               "use_CXX11": [True, False],
./recipes/ceres-solver/all/conanfile.py:24:               "use_schur_specializations": [True, False]}

./recipes/openscenegraph/all/conanfile.py:28:        "use_fontconfig": [True, False],

./recipes/libsodium/1.0.18/conanfile.py:22:        "use_soname" : [True, False],

./recipes/jansson/all/conanfile.py:18:        "use_urandom": [True, False],
./recipes/jansson/all/conanfile.py:19:        "use_windows_cryptoapi": [True, False]

./recipes/fruit/all/conanfile.py:18:               "use_boost": [True, False],

./recipes/gtsam/all/conanfile.py:20:               "use_quaternions": [True, False],

./recipes/date/all/conanfile.py:19:               "use_system_tz_db": [True, False],
./recipes/date/all/conanfile.py:20:               "use_tz_db_in_dot": [True, False]}

```

> - with_xxx / without_xxx

Author: ```suggestion
- with_<dependency> / without_<dependency>
```

> - use_xxx

Author: ```suggestion
- use_<feature>
```

the actual recipe code then may look like:

```py
    options = {"use_tzdb": [True, False]}
    default_options = {"use_tzdb": True}
```

```py
    options = {"enable_locales": [True, False]}
    default_options = {"enable_locales": True}
```

```py

>     options = {"with_threads": [True, False]}
>     default_options = {"with_threads": True}

Author: ```suggestion
    options = {"with_zlib": [True, False]}
    default_options = {"with_zlib": True}
```
